ql/src/java/org/apache/hadoop/hive/ql/stats/StatsUtils.java

```diff
@@ -1043,12 +1017,8 @@ public static List<ColStatistics> getTableColumnStats(
     }
     if (fetchColStats && !colStatsToRetrieve.isEmpty()) {
       try {
-        List<ColumnStatisticsObj> colStat;
-        if (table.isNonNative() && table.getStorageHandler().canProvideColStatistics(table)) {
-          colStat = table.getStorageHandler().getColStatistics(table);
-        } else {
-          colStat = Hive.get().getTableColumnStatistics(dbName, tabName, colStatsToRetrieve, false);
-        }
+        List<ColumnStatisticsObj> colStat = Hive.get().getTableColumnStatistics(
```

Comment:
it seems like  Hive.get().getTableColumnStatistics() is the final place where we decide whether to fetch col stats from metastore or fetch from storagehandler, which is fine if it's the final decision, but in this case this method's comment here is not accurate:
```
Get table level column statistics from metastore for needed columns
```
if we don't want to overthink, it's just:
```
Get table level column statistics for needed columns
```

Reply:
done